libevmasm/Assembly.cpp

```diff
@@ -1422,8 +1457,20 @@ LinkerObject const& Assembly::assembleEOF() const
 
 	ret.bytecode += m_auxiliaryData;
 
-	auto dataLength = ret.bytecode.size() - dataStart;
-	setBigEndianUint16(ret.bytecode, dataSectionSizeOffset, dataLength);
+	auto appendedDataAndAuxDataSize = ret.bytecode.size() - dataStart;
```

Comment:
This name is very misleading since `m_auxiliaryData` and EOF aux data are different things. This value does not include EOF aux data size, so a clearer name would be:

```suggestion
	auto preDeployDataSectionSize = ret.bytecode.size() - dataStart;
```

TBH, it's still not as clear as I'd like, because the spec for some odd reason decided to define `pre_deploy_data_size` as something different from `size(pre_deploy_data)`, instead of just giving it a name of it own, so names referring to it are inherently ambiguous. But still better than the current name.

Reply:
And a few more naming suggestions to make it easier to relate them to the spec, now that it's clearer to me what they actually mean:
- `dataLength` -> `preDeployAndStaticAuxDataSize`
- `originalDataOffset` -> `staticAuxDataOffset`
- `maxAuxDataLoadNOffset` + 32 -> `staticAuxDataSize`

And two more in `findMaxAuxDataLoadNOffset()` above:
- `i` -> `item`
- `result` -> `maxOffset`

Reply:
Done. Also added a couple of sentences explaining the logic.